crates/cast/src/lib.rs

```diff
@@ -344,10 +358,17 @@ pub async fn handle_wait_for_tx<T>(
     provider: &JsonRpcClient<HttpTransport>,
     transaction_hash: FieldElement,
     return_value: T,
-    wait: bool,
+    wait_config: WaitForTx,
 ) -> Result<T> {
-    if wait {
-        return match wait_for_tx(provider, transaction_hash, DEFAULT_RETRIES).await {
+    if wait_config.wait {
```

Comment:
If I understand correctly in the current solution when user passes `--wait-timeout` or `--wait-retry-interval` in a command without `--wait` flag, then both flags are ignored. Imo `--wait` flag should be used only to load wait config from Scarb.toml or to use default values.

Reply:
> If I understand correctly in the current solution when user passes --wait-timeout or --wait-retry-interval in a command without --wait flag, then both flags are ignored. 

yes, because we're not waiting for the txs

> Imo --wait flag should be used only to load wait config from Scarb.toml or to use default values.

not sure I follow - atm you can use just `--wait`, and there are default values for both timeout and retry (or are taken from Scarb.toml, if defined). You do not need to pass --wait, if you don't want to wait 👍

Reply:
I think we should accept flags like these at all times. We don't need rpc url when using `account create`, but are still able to provide it, just like with any other config values set in cast config.